src/plugins/explore/public/components/tabs/tabs.tsx

```diff
@@ -37,6 +37,12 @@ export const ExploreTabs = () => {
       dispatch(setActiveTab(selectedTab.id));
 
       const activeTab = services.tabRegistry.getTab(selectedTab.id);
+
+      if (activeTab?.onActive) {
+        activeTab?.onActive();
+      }
+      // TODO: put onInactive here? i mean the tabs are technically changing here as well
```

Comment:
yes lets add it. but lets call it before the new active tab is active and before all the dispatch, ie at the top of the function:

```
const onTabsClick = useCallback(
    (selectedTab: EuiTabbedContentTab) => {
        const previousActiveTab = services.tabRegistry.getTab(activeTabId);
        previousActiveTab?.onInactive?.();
        
        dispatch(setActiveTab(selectedTab.id))
        const newActiveTab = services.tabRegistry.getTab(selectedTab.id);
        newActiveTab?.onActive?.();

```

Reply:
oh sorry this todo was not meant to be here. i'll add this in.

Reply:
removed onActive + todo for onInactive